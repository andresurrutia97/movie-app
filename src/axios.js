import axios from 'axios';

const instance = axios.create({
  baseURL: 'https://api.themoviedb.org/3',
  headers: {
    Authorization: `Bearer ${process.env.REACT_APP_TOKEN}`,
  },
});

export default instance;
