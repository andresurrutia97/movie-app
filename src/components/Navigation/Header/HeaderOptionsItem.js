import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const HeaderOptionsItem = ({ path, name }) => {
  return (
    <NavLink className="flex items-center ml-6" to={path} exact>
      <span>{name}</span>
    </NavLink>
  );
};

HeaderOptionsItem.propTypes = {
  path: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default HeaderOptionsItem;
