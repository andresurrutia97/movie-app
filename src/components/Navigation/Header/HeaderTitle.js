import React from 'react';
import PropTypes from 'prop-types';

const HeaderTitle = ({ title }) => {
  return (
    <span className="font-semibold flex justify-center custom-blue-text text-2xl">
      {title}
    </span>
  );
};

HeaderTitle.propTypes = {
  title: PropTypes.string.isRequired,
};

export default HeaderTitle;
