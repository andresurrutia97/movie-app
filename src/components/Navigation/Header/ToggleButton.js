import React from 'react';
import PropTypes from 'prop-types';

const ToggleButton = ({ clicked }) => {
  return (
    <div className="block absolute left-0 md:hidden">
      <button
        onClick={clicked}
        type="button"
        className="flex items-center text-white"
      >
        <span className="material-icons">menu</span>
      </button>
    </div>
  );
};

ToggleButton.propTypes = {
  clicked: PropTypes.func.isRequired,
};

export default ToggleButton;
