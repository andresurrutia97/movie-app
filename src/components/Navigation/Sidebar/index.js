import React from 'react';
import PropTypes from 'prop-types';

import Routes from '../../../routes';
import Backdrop from '../../Ui/Backdrop';
import SidebarOptions from './SidebarOptions';
import logo from '../../../img/logoWeb.png';

const Sidebar = ({ open, openHandler }) => {
  const baseSidebarClass =
    'h-full sidebar-width bg-black fixed py-6 transition-transform duration-300 ease-in-out z-30 md:hidden';

  const sidebarClass = open
    ? [baseSidebarClass, 'sidebar-open']
    : [baseSidebarClass, 'sidebar-close'];

  return (
    <>
      <Backdrop show={open} closed={openHandler} />
      <div className={sidebarClass.join(' ')}>
        <div className="h-full flex flex-col items-center">
          <img className="w-48" src={logo} alt="logo" />
          <SidebarOptions close={openHandler} routes={Routes} />
        </div>
      </div>
    </>
  );
};

Sidebar.propTypes = {
  open: PropTypes.bool.isRequired,
  openHandler: PropTypes.func.isRequired,
};

export default Sidebar;
