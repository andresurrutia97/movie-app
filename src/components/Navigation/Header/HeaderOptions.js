import React from 'react';
import PropTypes from 'prop-types';

import HeaderOptionsItem from './HeaderOptionsItem';
import Dropdown from './UserDropdown/UserDropdown';

const HeaderOptions = ({ routes }) => {
  const options = routes.map(({ path, name }) => {
    return (
      <li key={path}>
        <HeaderOptionsItem path={path} name={name} />
      </li>
    );
  });

  return (
    <div className=" hidden md:ml-8 md:w-full md:block md:flex md:justify-between">
      <ul className="flex flex-col text-sm text-white md:flex-row md:items-center">
        {options}
      </ul>
      <Dropdown />
    </div>
  );
};

HeaderOptions.propTypes = {
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string,
      name: PropTypes.string,
    })
  ).isRequired,
};

export default HeaderOptions;
