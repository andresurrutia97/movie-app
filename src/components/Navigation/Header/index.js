import React, { useRef, useEffect } from 'react';
import PropTypes from 'prop-types';

import ToggleButton from './ToggleButton';
import HeaderTitle from './HeaderTitle';
import HeaderOptions from './HeaderOptions';
import Routes from '../../../routes';

const Header = ({ openSidebarHandler }) => {
  const [navBackground, setNavBackground] = React.useState(
    'custom-black-header-grad'
  );
  const navRef = useRef();

  useEffect(() => {
    const handleScroll = () => {
      const { scrollY } = window;
      if (scrollY >= 10) {
        setNavBackground('custom-black-header');
      } else {
        setNavBackground('custom-black-header-grad');
      }
    };
    document.addEventListener('scroll', handleScroll);
    return () => {
      document.removeEventListener('scroll', handleScroll);
    };
  }, []);

  return (
    <nav
      ref={navRef}
      className={`w-full h-16 fixed px-4 py-3 z-10 transition-colors duration-500 ${navBackground}`}
    >
      <div className="w-full flex justify-center items-center relative md:px-16 ">
        <ToggleButton clicked={openSidebarHandler} />
        <HeaderTitle title="Uruflix" />
        <HeaderOptions routes={Routes} />
      </div>
    </nav>
  );
};

Header.propTypes = {
  openSidebarHandler: PropTypes.func,
};

Header.defaultProps = {
  openSidebarHandler: () => {},
};

export default Header;
