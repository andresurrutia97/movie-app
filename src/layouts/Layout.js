import React from 'react';
import PropTypes from 'prop-types';

import { useDispatch } from '../utils/reduxHooks';
import { fetchUserInfo } from '../store/actions/UserActions';
import Header from '../components/Navigation/Header';
import Sidebar from '../components/Navigation/Sidebar';

const Layout = ({ children }) => {
  const dispatch = useDispatch();

  React.useEffect(() => {
    dispatch(fetchUserInfo());
  }, [dispatch]);

  const [openSidebar, setOpenSidebar] = React.useState(false);
  /**
   * handleOpenSidebar() controls the opening and
   * closing of the sidebar
   */
  const handleOpenSidebar = () => {
    setOpenSidebar(!openSidebar);
  };

  return (
    <div className="flex">
      <Header openSidebarHandler={handleOpenSidebar} />
      <Sidebar open={openSidebar} openHandler={handleOpenSidebar} />
      <div className="w-full h-full mt-16px md:mt-0">{children}</div>
    </div>
  );
};

Layout.propTypes = {
  children: PropTypes.node,
};

Layout.defaultProps = {
  children: null,
};

export default Layout;
