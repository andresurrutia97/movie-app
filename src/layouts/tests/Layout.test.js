import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import axios from 'axios';

import Layout from '../Layout';
import Header from '../../components/Navigation/Header';
import Sidebar from '../../components/Navigation/Sidebar';
import ToggleButton from '../../components/Navigation/Header/ToggleButton';

import * as ReduxHooks from '../../utils/reduxHooks';

jest.mock('axios');

describe('<Layout/>', () => {
  let wrapper;
  let store;
  const data = {
    data: {
      desc: 'Hi, im Andres and i like action movies and dogs',
      email: 'test@test.com',
      img: 'imgPath.jpg',
      name: 'Andres Urrutia Solarte f',
      tel: '3001231234',
    },
  };
  const component = (
    <BrowserRouter>
      <Layout />
    </BrowserRouter>
  );
  const mockSetState = jest.fn();

  beforeEach(() => {
    axios.get.mockResolvedValue(data);
    jest
      .spyOn(React, 'useState')
      .mockImplementation((init) => [init, mockSetState]);

    store = configureStore([thunk])({});

    jest.spyOn(React, 'useEffect').mockImplementationOnce((f) => f());
    jest
      .spyOn(ReduxHooks, 'useSelector')
      .mockImplementation(() => store.getState());
    jest
      .spyOn(ReduxHooks, 'useDispatch')
      .mockImplementation(() => store.dispatch);
    wrapper = mount(component);
  });

  afterEach(() => jest.clearAllMocks());

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain one instace of <Header/> and <Sidebar/>', () => {
    expect(wrapper.find(Header)).toHaveLength(1);
    expect(wrapper.find(Sidebar)).toHaveLength(1);
  });

  it('should call the setState function when handleOpenSidebar() is called', () => {
    expect(mockSetState).toBeCalledTimes(0);
    wrapper.find(Header).find(ToggleButton).find('button').simulate('click');
    expect(mockSetState).toBeCalledTimes(1);
  });

  it('should dispatch the fetchUserInfo action', () => {
    expect(store.getActions()).toEqual(
      expect.arrayContaining([{ type: 'FETCH_USER_INFO_START' }])
    );
    expect(store.getActions()).toEqual(
      expect.arrayContaining([
        { type: 'FETCH_USER_INFO_SUCCESS', userInfo: data.data },
      ])
    );
  });
});
