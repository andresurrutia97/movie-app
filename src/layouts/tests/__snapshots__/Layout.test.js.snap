// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> should render correctly  1`] = `
<div
  className="flex"
>
  <nav
    className="w-full h-16 fixed px-4 py-3 z-10 transition-colors duration-500 custom-black-header-grad"
  >
    <div
      className="w-full flex justify-center items-center relative md:px-16 "
    >
      <div
        className="block absolute left-0 md:hidden"
      >
        <button
          className="flex items-center text-white"
          onClick={[Function]}
          type="button"
        >
          <span
            className="material-icons"
          >
            menu
          </span>
        </button>
      </div>
      <span
        className="font-semibold flex justify-center custom-blue-text text-2xl"
      >
        Uruflix
      </span>
      <div
        className=" hidden md:ml-8 md:w-full md:block md:flex md:justify-between"
      >
        <ul
          className="flex flex-col text-sm text-white md:flex-row md:items-center"
        >
          <li>
            <a
              aria-current="page"
              className="flex items-center ml-6 active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              <span>
                Home
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current={null}
              className="flex items-center ml-6"
              href="/user"
              onClick={[Function]}
            >
              <span>
                Profile
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current={null}
              className="flex items-center ml-6"
              href="/toc"
              onClick={[Function]}
            >
              <span>
                ToC
              </span>
            </a>
          </li>
        </ul>
        <div
          className="relative"
        >
          <button
            className="relative z-10 block h-8 w-8 rounded-full overflow-hidden border-2 custom-blue-border focus:outline-none"
            onClick={[Function]}
            type="button"
          >
            <img
              alt="Your avatar"
              className="h-full w-full object-cover"
              src="default-user-image.png"
            />
          </button>
          <div
            className="hidden absolute right-0 mt-2 py-2 text-sm w-48 bg-white rounded custom-shadow"
          >
            <div
              className="flex justify-between itms-center text-gray-700 py-2 px-5 cursor-pointer"
              onClick={[Function]}
              role="presentation"
            >
              Profile
              <span
                className="material-icons"
              >
                person
              </span>
            </div>
            <div
              className="flex justify-between itms-center text-gray-700 py-2 px-5 cursor-pointer"
              onClick={[Function]}
              role="presentation"
            >
              <div>
                Logout
              </div>
              <span
                className="material-icons"
              >
                exit_to_app
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="fixed w-full h-full bg-black opacity-50 z-30 hidden"
    onClick={[Function]}
    role="presentation"
  />
  <div
    className="h-full sidebar-width bg-black fixed py-6 transition-transform duration-300 ease-in-out z-30 md:hidden sidebar-close"
  >
    <div
      className="h-full flex flex-col items-center"
    >
      <img
        alt="logo"
        className="w-48"
        src="logoWeb.png"
      />
      <ul
        className="w-full h-full mt-10 flex flex-col justify-between text-gray-300"
      >
        <div>
          <li
            className="item-sidebar"
            onClick={[Function]}
            role="presentation"
          >
            <a
              aria-current="page"
              className="flex items-center active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              <div
                className="flag"
              />
              <span
                className="ml-5"
              >
                Home
              </span>
            </a>
          </li>
          <li
            className="item-sidebar"
            onClick={[Function]}
            role="presentation"
          >
            <a
              aria-current={null}
              className="flex items-center"
              href="/user"
              onClick={[Function]}
            >
              <div
                className="flag"
              />
              <span
                className="ml-5"
              >
                Profile
              </span>
            </a>
          </li>
          <li
            className="item-sidebar"
            onClick={[Function]}
            role="presentation"
          >
            <a
              aria-current={null}
              className="flex items-center"
              href="/toc"
              onClick={[Function]}
            >
              <div
                className="flag"
              />
              <span
                className="ml-5"
              >
                ToC
              </span>
            </a>
          </li>
        </div>
        <li
          className="flex justify-between ml-2 itms-center text-gray-400 py-2 px-5 cursor-pointer"
          onClick={[Function]}
          role="presentation"
        >
          Logout
          <span
            className="material-icons"
          >
            exit_to_app
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="w-full h-full mt-16px md:mt-0"
  />
</div>
`;
